library/src/main/java/com/novoda/downloadmanager/lib/DownloadManagerModules.java

```diff
@@ -0,0 +1,101 @@
+package com.novoda.downloadmanager.lib;
+
+import android.content.Context;
+
+import com.novoda.downloadmanager.notifications.CancelledNotificationCustomiser;
+import com.novoda.downloadmanager.notifications.CompleteNotificationCustomiser;
+import com.novoda.downloadmanager.notifications.DownloadingNotificationCustomiser;
+import com.novoda.downloadmanager.notifications.FailedNotificationCustomiser;
+import com.novoda.downloadmanager.notifications.NotificationImageRetriever;
+import com.novoda.downloadmanager.notifications.QueuedNotificationCustomiser;
+
+public interface DownloadManagerModules {
+
+    NotificationImageRetriever getNotificationImageRetriever();
+
+    DownloadClientReadyChecker getDownloadClientReadyChecker();
+
+    QueuedNotificationCustomiser getQueuedNotificationCustomiser();
+
+    DownloadingNotificationCustomiser getDownloadingNotificationCustomiser();
+
+    CompleteNotificationCustomiser getCompleteNotificationCustomiser();
+
+    CancelledNotificationCustomiser getCancelledNotificationCustomiser();
+
+    FailedNotificationCustomiser getFailedNotificationCustomiser();
+
+    class Builder {
+
+        private final Context context;
+
+        private QueuedNotificationCustomiser queuedCustomiser;
```

Comment:
I am with @amlcurran , if the client passes null then we should instead throw a `DeveloperError` but using the default instead of what he's passing is breaking the contract. The `withBlablabla` is a promise we are making to him that we'll be using what he provides, even `null`.